docs/source/configuration.rst

```diff
@@ -107,7 +107,7 @@ steps overriding those from earlier:
 This whole structure leads to efficient configuration, in particular
 in projects which utilise a lot of complicated templating.
 
-.. _templateconfig:
+.. _ruleconfig:
```

Comment:
Error from #2291 where we inserted a section here, but didn't give it it's own title. I reinsert `_templateconfig` further down.